### PR TITLE
Update PETSc to 3.20.0

### LINKED ISSUE #25783: Update PETSc to 3.20.0
## Reason
The PETSc submodule is currently on 3.19.3 plus pre-release commits from their `main` branch. This was a temporary change to allow for [cutting edge improvements](https://gitlab.com/petsc/petsc/-/merge_requests/6642) in MOOSE. Now that these improvements are in a release, we should update to that release.

## Design
Update the PETSc submodule to 3.20.0.

## Impact
New features and bugfixes, better stability.

### PR DESCRIPTION
**Pull request by cticenhour**
Closes #25783 

CC: @lindsayad

### CODE REVIEW

**Comment by cticenhour** [comment]
@loganharbour Looks like we'll need to get CMake 3.21.0 on the containers: https://civet.inl.gov/job/1833298/

**Comment by loganharbour** [comment]
Just add `--download-cmake` to https://github.com/idaholab/moose/blob/1a588bd572fde1c4849fd6e1deb1bd2b67021e8c/apptainer/petsc.def#L67.

cmake comes from RHEL and we only get the current stable version for rocky 8.6 at the moment

**Comment by loganharbour** [comment]
These two diffs are new:

```
[53.93s]     DIFF mortar/continuity-3d-non-conforming.continuity_penalty_sphere_hex20 FAILED (EXODIFF) [FINISHED]
[56.86s]     DIFF mortar/continuity-3d-non-conforming.continuity_penalty_sphere_hex20_ad FAILED (EXODIFF) [FINISHED]
```

see https://civet.inl.gov/job/1835600/.

Mind helping @lindsayad or @recuero?

**Review by loganharbour — CHANGES_REQUESTED** [review]
Reminder for myself to finalize the 32bit container build and recipes before this gets in.

**Comment by lindsayad** [comment]
> Mind helping @lindsayad or @recuero?

yea no problem

**Comment by loganharbour** [comment]
Thanks. Feel free to push directly if you end up with a fix

**Review by loganharbour — DISMISSED** [review]

File: conda/libmesh/conda_build_config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 moose_petsc:
-  - moose-petsc 3.19.3.884.g08c1f94318e build_1
+  - moose-petsc 3.20.0 build_0
```
Comment (milljm) [inline conda/libmesh/conda_build_config.yaml:2]: Ahhh... thats better!

**Review by milljm — DISMISSED** [review]
This looks good! While we dropped moose-test-tools, which does affect moose-dev in a way, I think we're good to go as far as conda package dependencies roll.

**Review by milljm — DISMISSED** [review]

**Comment by roystgnr** [comment]
> sanity check on the skips in https://github.com/idaholab/moose/pull/25784 please; Jason will review the rest

The "[Limit IGA tests to 64bit dof indices](https://github.com/idaholab/moose/pull/25784/commits/ce1df363ff4ae60456b53eba475583def8cd4f3c)" commit has the flaw that it stabs to the core of my soul with a knife made of white-hot shame ... but engineering-wise it's probably the best we can do on short notice.  The updated CI here is exposing a preexisting bug, not adding a new one.

**Comment by loganharbour** [comment]
> it stabs to the core of my soul with a knife made of white-hot shame 

controlling-laugh-laughing

**Review by milljm — DISMISSED** [review]
🎉

**Review by lindsayad — APPROVED** [review]

**Pull request merged**